### DISCUSSION TITLE
Diarization Results are different from using API and Playground

### DISCUSSION THREAD

**Discussion by wonnoa** [deepgram/community#238] [General help] [answered]
### Which Deepgram product are you using?

Deepgram API

### Details

I am using deepgram api to transcribe.

****Problem =** 
****but diarization result is different from playground result even if i set the configurations and models same.
especially in terms of diarization. playground does the job better
api creates speaker 2 constantly even if in the audio, there are only two speakers; speaker 0 and 1**

**Question=
why does this happen and how can i make api result same as playground?****

results for identical audio

**### [api result] :** 
Speaker 0: Hello? Hello. Hey, Mark. How are you? Thanks for being with me.

How are you? I'm

Speaker 1: good. Mhmm. How about you?

Speaker 2: Yeah. Well, it's Friday. So that's that's a good news day. Alright. It's Friday.

Just to make sure, please, our our connections clear. No problem.

Speaker 1: Yeah. It's clear.

Speaker 2: Great. Thank you. Speaking speed good. 50 50 will try very natural conversation today. And, of course, any corrections, they will be here, any new words, they will be here.

Mhmm. And if you have any questions, Mark, just to ask. Yeah? But before we begin, I have a couple of materials, but I thought you were going on vacation.

Speaker 1: Oh, yeah. After work, I had been pecking my bags until now.

Speaker 2: Okay. I see. And when do you leave?

**### [playground result]:**

Speaker 0:

Hello? Hello. Hey, Mark. How are you? Thanks for being with me.

How are you? I'm good. Mhmm. How about you?

Speaker 1:

Yeah. Well, it's Friday. So that's that's

Speaker 0:

a good news day. Alright.

Speaker 1:

It's Friday. Just to make sure, please, our our connections clear. No problem.

Speaker 0:

Yeah. It's clear.

Speaker 1:

Great. Thank you. Speaking speed good. 50 50 will try very natural conversation today. And of course any corrections, they will be here, any new words, they will be here.

Mhmm. And if you have any questions, Mark, just to ask. Yeah? But before we begin, I have a couple of materials, but I thought you were going on vacation.

Speaker 0:

Oh, yeah. After work, I had been pecking my bags until now.

Speaker 1:

Okay. I see. And when do you leave?

### If possible, please attach your code or paste it into the text box.

this is my code = 
```
async def transcribe_file(filepath):
    # Create Deepgram client
    dg_client = Deepgram('REDACTED')

    # Read the audio file
    with open(filepath, 'rb') as audio:
        source = {'buffer': audio.read(), 'mimetype': 'audio/m4a'} 

    # Get the transcription
    response = await dg_client.transcription.prerecorded(source, 
    {'smart_format': True, 'punctuate': True, 'diarize': True, 'model':'nova', 'model_option': 'general', 'paragraphs': True})

    # Save the JSON response
    json_path = os.path.splitext(filepath)[0] + '.json'
    with open(json_path, 'w') as json_file:
        json.dump(response, json_file)
    print(f"JSON response saved to {json_path}")
```

**Comment by jjmaldonis** [comment]
Thanks @wonnoa for the report.

First, please regenerate your API key as soon as possible and delete your current one.

I would highly recommend setting your API key in your environment variables. A quick google should show you steps on how to do this. Then, in Python, you can safely load your API key into your code via `DEEPGRAM_API_KEY = os.environ["DEEPGRAM_API_KEY"]`.

Second, it's a bit strange to see these different results, but every request to Deepgram (even with the same audio and the same parameters) can result in a different transcript and/or diarization. In this case, it looks like those differences are noticeable. If the inputs are identical then there likely isn't anything we can do about this, and it was just coincidence that the results in playground were better. Or is this difference reproduceable?

If you are able to provide the request IDs for the two API calls (one in playground and one in python) that are showing the different results, I can dig into this more.

**Comment by wuzimi** [comment]
Same issue here. Still unsolved?

**Comment by jpvajda** [comment]
👋  I'm just checking in on this thread to see if this issue is still a problem.

**Comment by piegu** [comment]
> every request to Deepgram (even with the same audio and the same parameters) can result in a different transcript and/or diarization

Hi @jjmaldonis.

What you said (see quote) sounds like a possible problem. Can you explain the technical reason why "_every request to Deepgram (even with the same audio and the same parameters) can result in a different transcript and/or diarization_"?

And, is there a parameter to setup (like temperature = 0 for LLM) in order to get the exact same transcript and/or diarization from Deepgram (with the same audio and the same parameters as model)?

Thank you.
Pierre

**Reply by jpvajda** [reply, marked as answer]
@piegu  I think what @jjmaldonis  is speaking to is how AI models are non-deterministic in their results.  

To answer your other question - We don't currently offer a temperature parameter at this time. 

What would you do if we did offer that to solve this problem?

**Reply by piegu** [reply]
> What would you do if we did offer that to solve this problem?

I would use and pay the Deepgram API 👍
